ecs-agent/csiclient/csi_client_linux_integ_test.go

```diff
@@ -0,0 +1,91 @@
+//go:build linux && integration
+// +build linux,integration
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package csiclient
+
+import (
+	"context"
+	"encoding/json"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/api/attachment/resource"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const timeoutDuration = 1 * time.Second
+
+// Tests that CSIClient can fetch EBS volume metrics from EBS CSI Driver.
+//
+// This test assumes that there is only one EBS Volume attached to the host that's mounted
+// on "/" path. It also assumes that an EBS CSI Driver is running and listening on
+// /tmp/ebs-csi-driver.sock socket file.
+//
+// A typical environment for this test is an EC2 instance with a single EBS volume as its
+// root device volume.
+func TestNodeVolumeStats(t *testing.T) {
+	skipIfLsblkVerIsOld(t)
+
+	// Find the root volume using lsblk
+	lsblkCtx, lsblkCtxCancel := context.WithTimeout(context.Background(), timeoutDuration)
+	defer lsblkCtxCancel()
+	output, err := exec.CommandContext(lsblkCtx, "lsblk", "-o", "NAME,SERIAL", "-J").CombinedOutput()
+	require.NoError(t, err)
+	var lsblkOut resource.LsblkOutput
+	err = json.Unmarshal(output, &lsblkOut)
+	require.NoError(t, err)
+	require.True(t, len(lsblkOut.BlockDevices) > 0)
+
+	// There is no serial reported for EBS volumes on Xen devices.
+	// Skip the test for such instances.
+	volumeID := lsblkOut.BlockDevices[0].Serial
+	if volumeID == "" {
+		t.Skip("Test not supported on xen instances")
+	}
+
+	// Get metrics for the root volume from EBS CSI Driver.
+	csiClient := NewCSIClient("/tmp/ebs-csi-driver.sock")
+	getVolumeCtx, getVolumeCtxCancel := context.WithTimeout(context.Background(), timeoutDuration)
+	defer getVolumeCtxCancel()
+	metrics, err := csiClient.GetVolumeMetrics(getVolumeCtx, volumeID, "/")
```

Comment:
nit - IIRC the implementation of `GetVolumeMetrics` doesn't care the value of the `volumeID`. As long as it's not empty, it will pass. So w can also run this test on xen.

Reply:
Oh that'd be great. Let me try that.

Reply:
Oh you mean I can pass any random string? Let me try that.

Reply:
Right that works! I will update the PR. Thanks!

Reply:
Updated.

Reply:
Yes exactly!